release-notes/versioned/pulsar-3.1.0.md

```diff
@@ -0,0 +1,282 @@
+- Release EntryBuffer after parse proto object [#20170](https://github.com/apache/pulsar/pull/20170)
+- Remove useless field in the DelayedMessageIndexBucketSegment.proto [#20166](https://github.com/apache/pulsar/pull/20166)
+- Optimization protobuf code in the bucket delayed tracker [#20158](https://github.com/apache/pulsar/pull/20158)
+- Move bitmap from lastMutableBucket to ImmutableBucket [#20156](https://github.com/apache/pulsar/pull/20156)
+- Producer/Consumer should call allowTopicOperationAsync [#20142](https://github.com/apache/pulsar/pull/20142)
+- Fix entry filter feature for the non-persistent topic [#20141](https://github.com/apache/pulsar/pull/20141)
+- Optimize delayed metadata index bitmap [#20136](https://github.com/apache/pulsar/pull/20136)
+- Fast fix infinite HTTP call getSubscriptions caused by wrong topicName [#20131](https://github.com/apache/pulsar/pull/20131)
+- Make timer execute immediately after load index [#20126](https://github.com/apache/pulsar/pull/20126)
+- Cache LedgerHandle in BookkeeperBucketSnapshotStorage [#20117](https://github.com/apache/pulsar/pull/20117)
+- Pin AppendIndexMetadataInterceptor to field in `ManagedLedgerInterceptorImpl` [#20112](https://github.com/apache/pulsar/pull/20112)
+- Close temporary open ledger in BookkeeperBucketSnapshotStorage [#20111](https://github.com/apache/pulsar/pull/20111)
+- Fix use error TimeUnit to record publish latency [#20074](https://github.com/apache/pulsar/pull/20074)
+- Fix the reason label of authentication metrics [#20030](https://github.com/apache/pulsar/pull/20030)
+- Save createIfMissing in TopicLoadingContext [#19993](https://github.com/apache/pulsar/pull/19993)
+- Log resource usage rate of brokers that need to be offloaded in ThresholdShedder [#19983](https://github.com/apache/pulsar/pull/19983)
+- Fix getPartitionedStats miss subscription's messageAckRate [#19870](https://github.com/apache/pulsar/pull/19870)
+- Fix wrong logic of method TopicName.getPartition(int index) [#19841](https://github.com/apache/pulsar/pull/19841)
+- Fix `Update contains no change` error when use `--update-auth-data` flag to update function/sink/source [#19450](https://github.com/apache/pulsar/pull/19450)
+- Add `topic_load_failed` metric [#19236](https://github.com/apache/pulsar/pull/19236)
+- Support getStats/update partitioned topic with `-partition-` [#19235](https://github.com/apache/pulsar/pull/19235)
+- Do not filter system topic while shedding. [#18949](https://github.com/apache/pulsar/pull/18949)
+- If ledger lost, cursor mark delete position can not forward [#18620](https://github.com/apache/pulsar/pull/18620)
+- Do not grant permission for each partition to reduce unnecessary zk metadata [#18222](https://github.com/apache/pulsar/pull/18222)
+- Support cgroup v2 by using `jdk.internal.platform.Metrics` in Pulsar Loadbalancer [#16832](https://github.com/apache/pulsar/pull/16832)
+
+### Clients
```

Comment:
replicate with the client's release notes.. does it make more sense to add a link instead?

Reply:
fixed